File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    int res = 0;
public:
    int goodNodes(TreeNode* root) {
        if(!root){
            return 0;
        }
        
        dfs(root, root->val);
        return res;
        
    }
    void dfs(TreeNode* node, int curr){
        if(!node){
            return;
        }
        curr = max(curr, node->val);
        if (node->val >= curr){
            res++;
        }

        if (node->left)dfs(node->left, curr);
        if (node->right) dfs(node->right, curr);
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        maxP = 0

        for r in range(1,len(prices)):
            if prices[r]< prices[l]:
                l = r
                continue
            curr = prices[r] - prices[l]
            maxP = max(curr, maxP)


        return maxP

        